vm/rust/src/juno_state_reader.rs

```diff
@@ -193,15 +206,22 @@ pub struct ClassInfo {
 }
 
 pub fn class_info_from_json_str(raw_json: &str) -> Result<BlockifierClassInfo, String> {
-    let class_info: ClassInfo = serde_json::from_str(raw_json).map_err(|err| format!("failed parsing class info: {:?}", err))?;
-    let class_def = class_info.contract_class.to_string();    
-    
-    let class: ContractClass = if let Ok(class) = ContractClassV0::try_from_json_string(class_def.as_str()) {
-        class.into()
-    } else if let Ok(class) = ContractClassV1::try_from_json_string(class_def.as_str()) {
-        class.into()
-    } else {
-        return Err("not a valid contract class".to_string())
-    };
-    return Ok(BlockifierClassInfo::new(&class.into(), class_info.sierra_program_length, class_info.abi_length).unwrap());
+    let class_info: ClassInfo = serde_json::from_str(raw_json)
+        .map_err(|err| format!("failed parsing class info: {:?}", err))?;
+    let class_def = class_info.contract_class.to_string();
+
+    let class: ContractClass =
+        if let Ok(class) = ContractClassV0::try_from_json_string(class_def.as_str()) {
+            class.into()
+        } else if let Ok(class) = ContractClassV1::try_from_json_string(class_def.as_str()) {
+            class.into()
+        } else {
+            return Err("not a valid contract class".to_string());
```

Comment:
nit: not a valid contract class version?

Reply:
it's not always about different (unsupported) contract class version. Class definition itself may contains different errors (like unexpected type for correct field)